sympy/core/power.py

```diff
@@ -1735,11 +1735,11 @@ def mul(d1, d2):
                 raise NotImplementedError()
         if not d.is_positive:
             g = g.simplify()
+            _, d = g.leadterm(x)
```

Comment:
There is no need to compute `d` if `g` is zero. Hence this this line is better placed after the next two lines immediately before the test.